src/App.js

```diff
@@ -1,16 +1,67 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+// import ColorSelector from './components/ColorSelector';
+
 
 const App = () => {
+  const initialCopy = chatMessages.map((chat) => {
+    return {...chat};
+  })
+
+  const [chatData, setChatData] = useState(initialCopy);
+  const [likedCount, setLikedCount] = useState(0);
+  // const [selectColor, setSelectColor] = useState(black);
+
+  const heartWidget = likedCount !== 0 ? `${likedCount} ❤️s`: '';
+
+  const updateLike = (chatEntryId, updatedLike) => {
+    let newLikeCount = likedCount; 
+    const newChatData = chatData.map((chatEntry) => {
+      if (chatEntry.id !== chatEntryId){
+        return chatEntry
+      } else {
+        if (updatedLike) {
+          newLikeCount++;
+        }else {
+          newLikeCount--;
+        }
```

Comment:
[suggestion] it can be error-prone to try to keep track of counts like this as you handle related operations. It's usually more robust, if it's feasible to do it with reasonable performance, to tally up the sum of likes whenever we render the `chatData`. If you do that, then there's no way to accidentally forget to add/remove a like.